cmd/celestia-appd/cmd/modify_root_command_multiplexer.go

```diff
@@ -33,22 +32,41 @@ func modifyRootCommand(rootCommand *cobra.Command) {
 		panic(err)
 	}
 
-	var extraArgs []string
-	if v2UpgradeHeight != "" {
-		extraArgs = append(extraArgs, "--v2-upgrade-height="+v2UpgradeHeight)
+	var (
+		embeddedArgsFile string
+		embeddedArgs     EmbeddedArgs
+	)
+	// get the home directory from the flag
```

Comment:
[blocking] This comment is misleading because the code is getting the embedded arg flag not the home flag. Instead of correcting the comment, consider extracting a function with a descriptive name to eliminate the comment entirely.

```go
	args := os.Args
	embeddedArgsFile = getEmbeddedArgsFile(args)
```

and later on

```go
func getEmbeddedArgsFile(args []string) string {
	for i := 0; i < len(args); i++ {
		if args[i] == fmt.Sprintf("--%s", FlagEmbeddedArgsFile) && i+1 < len(args) {
			return filepath.Clean(args[i+1])
		} else if strings.HasPrefix(args[i], fmt.Sprintf("--%s=", FlagEmbeddedArgsFile)) {
			return filepath.Clean(args[i][7:])
		}
	}
	return ""
}
```